reactioncommerce/reaction
# refactor: use functionsOfType to register multiple auth permissions functions

Impact: **minor**  
Type: **feature|refactor**

## Issue
Reaction currently allows only one active `getHasPermissionFunctionForUser` function to be registered, therefore only allowing one authorization service check.

## Solution
Update `context.userHasPermission` to allow multiple authorization services to be registered and checked against.

## Breaking changes
None.

## Testing
1. Start the app
1. See that the existing `legacy-authorization` package works as is expected
1. Further expansion on this will come in more auth-pod work
